### ISSUE TITLE
Postgres source `timestamp without time zone` is being stored as `timestamp with time zone` in Redshift Destination

### ISSUE THREAD

**Issue by rohaldb** [airbytehq/airbyte#45616]
### Connector Name

source-postgres

### Connector Version

3.6.18

### What step the error happened?

During the sync

### Relevant information

I have a field, called `timestamp` on one of my tables in postgres. As you can see here, it is of time `timestamp without timezone`

<img width="446" alt="Screenshot 2024-09-17 at 10 45 48 am" src="https://github.com/user-attachments/assets/8988c7a4-e8e8-40f9-bf5a-95eb3632f774">

<img width="518" alt="Screenshot 2024-09-17 at 10 46 00 am" src="https://github.com/user-attachments/assets/73195e5d-6b45-465b-8e0b-40e9a629cce7">

Before today, it was being synced to Redshift (v3.4.4) as a timestamp without timezone correctly. However, overnight I noticed that it is now being stored in redshift as a timestamp with timezone. See below for two 

<img width="553" alt="Screenshot 2024-09-17 at 10 47 25 am" src="https://github.com/user-attachments/assets/d31420a4-021d-44a3-940b-4553ad42810e">
<img width="887" alt="Screenshot 2024-09-17 at 10 48 18 am" src="https://github.com/user-attachments/assets/f05fc845-42ad-42e0-8be6-4fb03ef6f887">

The change in data type has broken my upstream DBT because in redshift you need to use different comparison operators for timestamps with/without timezone

The specific connection I'm using for debugging purposes it https://cloud.airbyte.com/workspaces/f72c2586-d563-401e-a87e-fb8d105f1239/connections/6a5e6e53-7bfe-43e9-99a2-833ab4174e31/replication

The table is called `InsightVersion` and the field (as mentioned) is called `timestamp`

### Relevant log output

```shell
I believe the exact sync where this change happened was https://cloud.airbyte.com/workspaces/f72c2586-d563-401e-a87e-fb8d105f1239/connections/6a5e6e53-7bfe-43e9-99a2-833ab4174e31/timeline?eventId=2347d0d3-9598-463e-b857-dc3e59d8eab8&openLogs=true but I couldn't parse it to find an issue as it is too large
```

### Contribute

- [ ] Yes, I want to contribute

**Comment by mathieu-luciani** [comment]
Same issue with destination-bigquery: since yesterday, the `timestamp without time zone` fields coming from postgres are being stored as `timestamp` in BigQuery.
Before that, they were stored as `datetime`

**Comment by rohaldb** [comment]
In a bizarre turn of events, this has reverted in my most recent sync back to timestamp without timezone. The exact sync can be found [here](https://cloud.airbyte.com/workspaces/f72c2586-d563-401e-a87e-fb8d105f1239/connections/6a5e6e53-7bfe-43e9-99a2-833ab4174e31/timeline?eventId=e63322e7-a55a-4bce-b9de-3c6badb752e3&openLogs=true). 

Has anyone else experienced this as well?

**Comment by mathieu-luciani** [comment]
Yep, same here: in our bigquery implementation, the `timestamp without timezone` fields from postgres have reverted back to `datetime`.

However, I'm afraid this doesn't mean we're out of the woods just yet: the same had happened to us last Friday where:
- at 18:30 UTC, the fields have been switched to `timestamp with timezone` (from `timestamp without timezone`), resulting in `timestamp` in BigQuery
`2024-09-13 18:33:14 destination > INFO sync-operations-6 i.a.i.b.d.o.AbstractStreamOperation(prepareFinalTable):290 Executing soft-reset on final table of stream StreamConfig(id=StreamId(finalNamespace=raw_prod, finalName=role, rawNamespace=airbyte_internal, rawName=raw_prod_raw__stream_role, originalNamespace=raw_prod, originalName=role), postImportAction=DEDUPE, primaryKey=[ColumnId(name=id, originalName=id, canonicalName=id)], cursor=Optional.empty, columns={ColumnId(name=id, originalName=id, canonicalName=id)=INTEGER, ColumnId(name=name, originalName=name, canonicalName=name)=STRING, ColumnId(name=created, originalName=created, canonicalName=created)=TIMESTAMP_WITH_TIMEZONE, ColumnId(name=updated, originalName=updated, canonicalName=updated)=TIMESTAMP_WITH_TIMEZONE, ColumnId(name=description, originalName=description, canonicalName=description)=STRING}, generationId=0, minimumGenerationId=0, syncId=17957485)`
- at 19:30 UTC, the fields have been switched back to `timestamp without timezone`, resulting in `datetime` in BiqQuery
`2024-09-13 19:34:35 destination > INFO sync-operations-7 i.a.i.b.d.o.AbstractStreamOperation(prepareFinalTable):290 Executing soft-reset on final table of stream StreamConfig(id=StreamId(finalNamespace=raw_prod, finalName=role, rawNamespace=airbyte_internal, rawName=raw_prod_raw__stream_role, originalNamespace=raw_prod, originalName=role), postImportAction=DEDUPE, primaryKey=[ColumnId(name=id, originalName=id, canonicalName=id)], cursor=Optional.empty, columns={ColumnId(name=id, originalName=id, canonicalName=id)=INTEGER, ColumnId(name=name, originalName=name, canonicalName=name)=STRING, ColumnId(name=created, originalName=created, canonicalName=created)=TIMESTAMP_WITHOUT_TIMEZONE, ColumnId(name=updated, originalName=updated, canonicalName=updated)=TIMESTAMP_WITHOUT_TIMEZONE, ColumnId(name=description, originalName=description, canonicalName=description)=STRING}, generationId=0, minimumGenerationId=0, syncId=17959960)`

Considering that the Postgres source connector hasn't been updated in 19 days (i.e. an update in the connector doesn't explain the issue or its resolution), it probably means that the issue we have seen over the last few days can reoccur at any time.
